### DISCUSSION TITLE
Get values from array on node? (only get first value)

### DISCUSSION THREAD

**Discussion by KenSense** [FreeOpcUa/opcua-asyncio#767] [Q&A]
This node works
_node = client.get_node(ua.NodeId('/Channel/Parameter/R', 2))_
When I ask the value I only get the first value of an array.  _var = await node.get_value()_

According to the machine documentation I should request it with this address:
_'/Channel/Parameter/R[u1,1]'_ or even as array of 10 parameters  _'/Channel/Parameter/R[u1,1,10]'_
Page 47 https://cache.industry.siemens.com/dl/files/318/109760318/att_960876/v1/840Dsl_OPCUA_commiss_man_0818_en-US.pdf

But /Channel/Parameter/R[u1,1] as node will always give me: 
_asyncua.ua.uaerrors._auto.BadNodeIdInvalid: "The syntax of the node id is not valid."_

So how can I access this data or does this library not support this?

Thanks

**Comment by AndreasHeine** [comment]
"asyncua.ua.uaerrors.auto.BadNodeIdInvalid" is a response from the opc ua server (siemens)!

**Reply by AndreasHeine** [reply]
can you post a screenshot from UA Expert like in the docs 
-> 
5.3 Variable access
5.3.1 Variable paths for NC access operations

**Reply by AndreasHeine** [reply]
> 5.2 Address space model
> Address space model
> If the OPC UA server is browsed, the available address space is mapped under the "Sinumerik"
> node.
> Global User Data (GUD) can be found under the "/Sinumerik/GUD" node.
> The PLC blocks (inputs, outputs, bit memory, data blocks) can be found under the "/Sinumerik/
> Plc" node.
> Machine data can be found under the node "/Sinumerik/TEA".
> Setting data can be found under the node "/Sinumerik/SEA".
> Observe the following while browsing:
> ● In the address space of the NC, the displayed variables always represent only the first
> parameter of the corresponding unit.
> Example:
> The R parameters can be found under "Sinumerik > Channel > Parameter > R". The
> corresponding identifier is called "/Channel/Parameter/R", which is finally mapped to "/
> Channel/Parameter/R[u1, 1]". If you want to access other parameters, you need to specify
> the corresponding index in brackets, e. g. "/Channel/Parameter/R[u2,56]".
> ● In the address space of the PLC, the displayed variables represent the access format that
> has to be extended accordingly.
> Example:
> The variable "/Plc/MB" is in the address space and is mapped to "/Plc/MB0". For accessing
> further bytes this variable must be extended by the appropriate byte number, e.g. "/Plc/
> MB6".
> ● The address space of the NC also contains variables that are not available in a
> corresponding machine configuration. These variables return "BadAttributeIdInvalid" as
> value.

from that i would guess it does not exist in the opc ua servers addressspace!

**Comment by KenSense** [comment]
It does show that there is no type provided for all nodes, not sure if that's ok
![DataAccess_UaExpert](https://user-images.githubusercontent.com/49442805/149356918-30b29f01-6759-4418-ab86-d5112c832173.jpg)

**Reply by AndreasHeine** [reply]
please post the jpg into the writing area (it will be uploaded)! i dont like to click on those weird links ^^

**Reply by KenSense** [reply]
So you think the server isn't working as it is documented?

**Reply by AndreasHeine** [reply]
hm it (R) has no TypeDefinition Reference and no DataType Attribute ??? this is weird!

**Reply by AndreasHeine** [reply]
what is underneath R if you unfold?

**Reply by KenSense** [reply]
> what is underneath R if you unfold?

Nothing R is the end

**Reply by KenSense** [reply]
Problem solved, the SINUMERIK OPC UA server on the machine needed a software update. (bug fix)
